.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
I think we should let it run for any PR, not only the ones based in `main`.

Besides, we should change the base of the PR to `dev`.